src/lib/types/option.ts

```diff
@@ -53,10 +55,16 @@ export enum IAuthType {
     NONE = 'none',
 }
 
+export type CustomAuthHandler = (
+    app: Express,
+    config: IUnleashConfig,
+    services: IUnleashServices,
```

Comment:
```suggestion
    config: Partial<IUnleashConfig>,
    services?: IUnleashServices,
```
Maybe making services optional in case you're an open source you can just ignore receiving it. 

While the config we're making it partial so you have to explicitly check for undefined when accessing the config and we can also send `{}` empty object without getting a compilation error